tests/components/plex/test_browse_media.py

```diff
@@ -77,10 +78,61 @@ async def test_browse_media(hass, hass_ws_client):
     result = msg["result"]
     assert result[ATTR_MEDIA_CONTENT_TYPE] == "server"
     assert result[ATTR_MEDIA_CONTENT_ID] == DEFAULT_DATA[CONF_SERVER_IDENTIFIER]
-    assert len(result["children"]) == len(mock_plex_server.library.sections())
+    assert len(result["children"]) == len(mock_plex_server.library.sections()) + len(
+        SPECIAL_METHODS
+    )
 
     tvshows = next(iter(x for x in result["children"] if x["title"] == "TV Shows"))
     playlists = next(iter(x for x in result["children"] if x["title"] == "Playlists"))
+    special_keys = list(SPECIAL_METHODS.keys())
+
+    # Browse into a special folder (server)
+    msg_id += 1
+    await websocket_client.send_json(
+        {
+            "id": msg_id,
+            "type": "media_player/browse_media",
+            "entity_id": media_players[0],
+            ATTR_MEDIA_CONTENT_TYPE: "server",
+            ATTR_MEDIA_CONTENT_ID: f"{DEFAULT_DATA[CONF_SERVER_IDENTIFIER]}:{special_keys[0]}",
+        }
+    )
+
+    msg = await websocket_client.receive_json()
+    assert msg["id"] == msg_id
+    assert msg["type"] == TYPE_RESULT
+    assert msg["success"]
+    result = msg["result"]
+    assert result[ATTR_MEDIA_CONTENT_TYPE] == "server"
+    assert (
+        result[ATTR_MEDIA_CONTENT_ID]
+        == f"{DEFAULT_DATA[CONF_SERVER_IDENTIFIER]}:{special_keys[0]}"
+    )
+    assert len(result["children"]) == len(mock_plex_server.library.onDeck())
+
+    # Browse into a special folder (library)
+    msg_id += 1
+    library_section_id = next(iter(mock_plex_server.library.sections())).key
+    await websocket_client.send_json(
+        {
+            "id": msg_id,
+            "type": "media_player/browse_media",
+            "entity_id": media_players[0],
+            ATTR_MEDIA_CONTENT_TYPE: "library",
+            ATTR_MEDIA_CONTENT_ID: f"{library_section_id}:{special_keys[1]}",
+        }
+    )
+
+    msg = await websocket_client.receive_json()
+    assert msg["id"] == msg_id
+    assert msg["type"] == TYPE_RESULT
+    assert msg["success"]
+    result = msg["result"]
+    assert result[ATTR_MEDIA_CONTENT_TYPE] == "library"
+    assert result[ATTR_MEDIA_CONTENT_ID] == f"{library_section_id}:{special_keys[1]}"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6103615)